tests/sagemaker-tests/test_sagemaker_utils.py

```diff
@@ -0,0 +1,193 @@
+            assert 'model.bin' in results and results['model.bin'] == True
+            assert 'test.yaml' in results and results['test.yaml'] == True
+            assert 'test.json' in results and results['test.json'] == True
+
+    # test case 2: abnormal cases
+    #     2.1: missing one of the four files
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        entry_path = os.path.join(tmpdirname, 'nc_infer_entry.py')
+        model_path = os.path.join(tmpdirname, 'model.bin')
+        yaml_path = os.path.join(tmpdirname, 'test.yaml')
+        json_path = os.path.join(tmpdirname, 'test.json')
+
+        output_path = os.path.join(tmpdirname, 'output_folder')
+        os.makedirs(output_path, exist_ok=True)
+
+        # not create the entry point file
+        create_dummy_file(model_path)
+        create_dummy_file(yaml_path)
+        create_dummy_file(json_path)
+        with pytest.raises(AssertionError, match='SageMaker entry point .* not exist'):
+            wrap_model_artifacts(model_path, yaml_path, json_path, entry_path,
+                                 output_path=output_path, output_tarfile_name='model')
+
+        os.remove(model_path)
+        create_dummy_file(entry_path)
+        # not create the model.bin file.
+        create_dummy_file(yaml_path)
+        create_dummy_file(json_path)
+        with pytest.raises(AssertionError, match='model file, .* not exist'):
+            wrap_model_artifacts(model_path, yaml_path, json_path, entry_path,
+                                 output_path=output_path, output_tarfile_name='model')
+
+        os.remove(yaml_path)
+        create_dummy_file(entry_path)
+        create_dummy_file(model_path)
+        # not create the test.yaml file.
+        create_dummy_file(json_path)
+        with pytest.raises(AssertionError, match='YAML .* not exist'):
+            wrap_model_artifacts(model_path, yaml_path, json_path, entry_path,
+                                 output_path=output_path, output_tarfile_name='model')
+
+        os.remove(json_path)
+        create_dummy_file(entry_path)
+        create_dummy_file(model_path)
+        create_dummy_file(yaml_path)
+        # not create the test.json file.
+        with pytest.raises(AssertionError, match='JSON .* not exist'):
+            wrap_model_artifacts(model_path, yaml_path, json_path, entry_path,
+                                 output_path=output_path, output_tarfile_name='model')
+
+    #     2.2: missing output folder
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        entry_path = os.path.join(tmpdirname, 'nc_infer_entry.py')
+        model_path = os.path.join(tmpdirname, 'model.bin')
+        yaml_path = os.path.join(tmpdirname, 'test.yaml')
+        json_path = os.path.join(tmpdirname, 'test.json')
+
+        create_dummy_file(entry_path)
+        create_dummy_file(model_path)
+        create_dummy_file(yaml_path)
+        create_dummy_file(json_path)
+        with pytest.raises(TypeError):
```

Comment:
Why this is a TypeError?

Reply:
Because missing of required argument error is TypeError.